vsphere/resource_vsphere_host_virtual_switch_test.go

```diff
@@ -0,0 +1,254 @@
+				CheckDestroy: testAccResourceVSphereHostVirtualSwitchExists(false),
+				Steps: []resource.TestStep{
+					{
+						Config: testAccResourceVSphereHostVirtualSwitchConfigStandbyLink(),
+						Check: resource.ComposeTestCheckFunc(
+							testAccResourceVSphereHostVirtualSwitchExists(true),
+						),
+					},
+				},
+			},
+		},
+		{
+			"basic, then change to standby with failover order",
+			resource.TestCase{
+				PreCheck: func() {
+					testAccPreCheck(tp)
+					testAccResourceVSphereHostVirtualSwitchPreCheck(tp)
+				},
+				Providers:    testAccProviders,
+				CheckDestroy: testAccResourceVSphereHostVirtualSwitchExists(false),
+				Steps: []resource.TestStep{
+					{
+						Config: testAccResourceVSphereHostVirtualSwitchConfig(),
+						Check: resource.ComposeTestCheckFunc(
+							testAccResourceVSphereHostVirtualSwitchExists(true),
+						),
+					},
+					{
+						Config: testAccResourceVSphereHostVirtualSwitchConfigStandbyLink(),
+						Check: resource.ComposeTestCheckFunc(
+							testAccResourceVSphereHostVirtualSwitchExists(true),
+						),
+					},
+				},
+			},
+		},
+	}
+
+	for _, tc := range testAccResourceVSphereHostVirtualSwitchCases {
+		t.Run(tc.name, func(t *testing.T) {
+			tp = t
+			resource.Test(t, tc.testCase)
+		})
+	}
+}
+
+func testAccResourceVSphereHostVirtualSwitchPreCheck(t *testing.T) {
+	if os.Getenv("VSPHERE_HOST_NIC0") == "" {
+		t.Skip("set VSPHERE_HOST_NIC0 to run vsphere_host_virtual_switch acceptance tests")
+	}
+	if os.Getenv("VSPHERE_HOST_NIC1") == "" {
+		t.Skip("set VSPHERE_HOST_NIC1 to run vsphere_host_virtual_switch acceptance tests")
+	}
+	if os.Getenv("VSPHERE_ESXI_HOST") == "" {
+		t.Skip("set VSPHERE_ESXI_HOST to run vsphere_host_virtual_switch acceptance tests")
+	}
+}
+
+func testAccResourceVSphereHostVirtualSwitchExists(expected bool) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources["vsphere_host_virtual_switch.switch"]
+		if !ok {
+			return errors.New("vsphere_host_virtual_switch.switch not found in state")
+		}
+
+		client := testAccProvider.Meta().(*govmomi.Client)
+		id := rs.Primary.ID
+		host := os.Getenv("VSPHERE_ESXI_HOST")
+		datacenter := os.Getenv("VSPHERE_DATACENTER")
+		timeout := time.Minute * 5
+		_, err := hostVSwitchFromName(client, id, host, datacenter, timeout)
+		if err != nil {
+			if err.Error() == fmt.Sprintf("vSwitch %s not found on host %s", id, host) && !expected {
+				// Expected missing
+				return nil
+			}
+			return err
+		}
+		if !expected {
+			return fmt.Errorf("expected vSwitch %s to still be missing", id)
+		}
+		return nil
+	}
+}
+
+func testAccResourceVSphereHostVirtualSwitchConfig() string {
+	return fmt.Sprintf(`
+variable "esxi_host" {
+  type    = "string"
+  default = "%s"
+}
+
+variable "datacenter" {
+  type    = "string"
+  default = "%s"
+}
+
+variable "host_nic0" {
+  type    = "string"
+  default = "%s"
+}
+
+variable "host_nic1" {
+  type    = "string"
+  default = "%s"
+}
```

Comment:
_minor_ we can remove the `type` from these

Reply:
Hmm, I thought that it's been generally recommended to use explicit variable type declaration in TF code ever since functionality for it was released in 0.6.10. Was I mistaken in that? I can change it, although it will take some getting used to for me writing my configs with some variables with defaults being implicit and the ones without defaults having `types`.

Like this just seems a little inconsistent to me:

```
variable "has_default" {
  default = "implied"
}

variable "no_default" {
  type = "string"
}
```

Reply:
> Was I mistaken in that? 

I'm not sure, from what I can see:

> If no default is provided, the type is assumed to be string.

it sounds like we can rely on that assumption long-term. There's no issue with leaving these in, but `go vet` will fail if these aren't strings, given they're marked as `%s` - so I think we could safely remove it (but it's no big deal)

Reply:
Yeah I can't remember 100%. IMO, it's not a matter of whether or not that's going to change longer term for me, it's more a matter of what looks consistent for me and what matches my personal style.

Been thinking a bit more about this actually, and I think changing this kind of goes with my general philosophy when writing TF code, which is mainly treating it like how I write go code, and we have implicit types in value declarations there so why not here as well? So will change this now :smile: